clang-tools-extra/clangd/AST.cpp

```diff
@@ -144,8 +144,13 @@ getQualification(ASTContext &Context, const DeclContext *DestContext,
   // since we stored inner-most parent first.
   std::string Result;
   llvm::raw_string_ostream OS(Result);
-  for (const auto *Parent : llvm::reverse(Parents))
+  for (const auto *Parent : llvm::reverse(Parents)) {
+    if (Parent != *Parents.rbegin() && Parent->isDependent() &&
+        Parent->getAsRecordDecl() &&
+        Parent->getAsRecordDecl()->getDescribedClassTemplate())
```

Comment:
Save the duplicate call to `getAsRecordDecl` by adding a variable? (+-)

Reply:
The function doesn't look expensive, and this would complicate the code, so I'd rather not.

Reply:
Fine with me